src/Voron/StorageEnvironment.cs

```diff
@@ -140,6 +140,9 @@ public StorageEnvironment(StorageEnvironmentOptions options)
                 _validPages[_validPages.Length - 1] |= unchecked(((long)ulong.MaxValue << (int)remainingBits));
 
                 _decompressionBuffers = new DecompressionBuffersPool(options);
+
+                options.InvokeOnCreateDirectory();
```

Comment:
Shouldn't we call it _only_ if we create a new database?

https://github.com/ravendb/ravendb/blob/412d9c18257789e0012b0673e50635c345c3056d/src/Voron/StorageEnvironment.cs#L155

Reply:
Yes, argument would be very helpful. I think we really need to allow users to distinguish if _create_ or _open_.

Given when it's called I wonder if `Storage.OnCreateDirectory...` isn't a bit confusing name. Maybe just `Storage.OnInitialize...` or `Storage.OnDirectoryInitialize...`. What do you think?

Reply:
Changing the name is better.
And the script can _check_ if the directories already exists, we don't need to provide that.

Reply:
OK, Storage.OnInitialize sounds good